src/components/tl/cuda/bcast/bcast_linear.c

```diff
@@ -0,0 +1,303 @@
+                    return;
+                }
+            }
+            task->bcast_linear.stage = STAGE_COPY;
+            if (task->bcast_linear.step < task->bcast_linear.num_steps) {
+                // go to next iteration
+                task->bcast_linear.stage = STAGE_COPY;
+                return;
+            } else {
+                // finish
+                task->bcast_linear.stage = STAGE_DONE;
+            }
+        case STAGE_DONE:
+            task->super.status = UCC_OK;
+            break;
+        default:
+            break;
+        }
+    } else {
+        // clients
+        // fall-through between cases is intentional
+        switch (task->bcast_linear.stage) {
+        case STAGE_WAIT_ROOT:
+            if (get_rank_step(task, task->bcast_linear.root, 0) >
+                task->bcast_linear.step) {
+                task->bcast_linear.stage = STAGE_CLIENT_COPY;
+                break;
+            } else {
+                return;
+            }
+        case STAGE_CLIENT_COPY:
+            // need to copy from root's scratch buffer
+            dbuf = PTR_OFFSET(task->bcast_linear.sbuf, offset_buff);
+            sbuf = PTR_OFFSET(TASK_SCRATCH(task, task->bcast_linear.root),
+                              task->bcast_linear.step % 2 * chunk_size);
+            st   = ecopy(dbuf, sbuf, chunk_size, exec,
+                       &task->bcast_linear.exec_task);
+            if (st != UCC_OK) {
+                ucc_error("failed to post ecopy task at client");
+                task->super.status = st;
+                return;
+            }
+            task->bcast_linear.stage = STAGE_CLIENT_COPY_WAIT;
+        case STAGE_CLIENT_COPY_WAIT:
+            etask = task->bcast_linear.exec_task;
+            if (etask) {
+                st = ucc_ee_executor_task_test(etask);
+                if (st == UCC_OK) {
+                    ucc_ee_executor_task_finalize(etask);
+                    task->bcast_linear.exec_task = NULL;
+                    ++task->bcast_linear.step;
+                    set_rank_step(task, trank, task->bcast_linear.step, 0);
+                    if (task->bcast_linear.step <
+                        task->bcast_linear.num_steps) {
+                        task->bcast_linear.stage = STAGE_WAIT_ROOT;
+                        return;
+                    } else {
+                        task->bcast_linear.stage = STAGE_DONE;
+                    }
+                } else {
+                    return;
+                }
+            } else {
+                return;
+            }
+        case STAGE_DONE:
+            task->super.status = UCC_OK;
+            break;
+        default:
+            break;
+        }
+    }
+}
+
+ucc_status_t ucc_tl_cuda_bcast_linear_start(ucc_coll_task_t *coll_task)
+{
+    ucc_tl_cuda_task_t *task = ucc_derived_of(coll_task, ucc_tl_cuda_task_t);
+    ucc_tl_cuda_team_t *team = TASK_TEAM(task);
+    ucc_coll_args_t *   args = &TASK_ARGS(task);
```

Comment:
alignment of "*", here and in other places

Reply:
Agree, but I use provided clang-format and it did this style, maybe it depends on clang-format version?